### ISSUE TITLE
[layers] 1-layer pre-LN Transformer encoder block (attention + FFN)

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5661]
## Summary

Add a `TransformerEncoderBlock` layer to `src/odyssey/core/layers/` — a single
Transformer encoder block (Vaswani et al. 2017, "Attention Is All You Need",
arXiv:1706.03762) in the **pre-norm (pre-LN)** arrangement:

```
u = x + MultiHeadAttention(LayerNorm_1(x))     # self-attention sub-layer
y = u + FeedForward(LayerNorm_2(u))            # position-wise FFN sub-layer
```

Batch-first `[batch, seq, d_model]`, shape-preserving, full-sequence `forward()`.
This is the composite primitive requested by the predictive-coding research track
(mvillmow/Random#51, "[ARCH-01] 1-layer Transformer block") — the Odyssey side is
the reusable library layer; the PC/BP sweep wiring lives downstream in the
predictive-coding-mojo repo.

## Design

- **Thin composition, no reimplementation.** Reuses the existing
  `MultiHeadAttention` (self-attention), `LayerNorm`, and `FeedForward` Modules
  verbatim, wiring them into the two pre-norm residual sub-layers and forwarding
  their parameters. Mirrors `MLPMixerBlock`'s reuse posture (`mlp_mixer.mojo`).
- **Pre-LN placement** (LayerNorm on each sub-layer's *input*) per Xiong et al.
  2020 (arXiv:2002.04745) — the modern default and the variant pinned by the
  tracking issue. This differs from the original post-LN Transformer.
- `MultiHeadAttention.forward` consumes the 3D sequence directly; `LayerNorm`
  and `FeedForward` are position-wise 2D ops, applied by flattening
  `[B, S, d_model] -> [B*S, d_model]` and reshaping back.
- `causal` forwards to the attention sub-layer's autoregressive `-inf` mask, so
  the same block serves encoder (non-causal) and decoder-style (causal) use.
- No BatchNorm anywhere; LayerNorm only. Activation / epsilon / mask conventions
  are inherited unchanged from the reused sub-modules.

## Acceptance criteria

- [ ] `TransformerEncoderBlock[dtype]` is `Module`-conforming (forward,
      parameters, train, eval); float32 default dtype.
- [ ] Constructor validates `d_model > 0`; `num_heads` divisibility inherited
      from the attention sub-layer; `d_ff <= 0` resolves to `4 * d_model`.
- [ ] `forward` preserves `[batch, seq, d_model]` and rejects non-3D /
      last-dim-mismatch inputs.
- [ ] `parameters()` returns 16 tensors (norm1 x2, attn x8, norm2 x2, ffn x4).
- [ ] PyTorch parity to 1e-5 for BOTH no-mask and causal cases, from a committed
      ramp-seeded f64 generator (regenerate + diff before commit).
- [ ] Public export via `core/layers/__init__.mojo`; primitive-test registration
      (`run_primitive_test.sh transformer`) + README table row.
- [ ] `--Werror`-clean; mojo format; ruff; markdownlint; pre-commit on touched
      files all green.

## Dependencies

- **Requires `MultiHeadAttention`** (PR #5640, branch `feat/attention-layer`),
  which is not yet on `origin/main`. This is a **stacked PR** — #5640 must merge
  first. `LayerNorm` (#5613) and `FeedForward` (#5607) are already on main.

## Reference

Vaswani, A., et al. (2017). Attention Is All You Need. NeurIPS 2017.
arXiv:1706.03762, §3.1-3.3. Pre-LN placement: Xiong, R., et al. (2020),
arXiv:2002.04745.
